### ISSUE TITLE
Test failure with OpenMPI 5.0.1

### ISSUE THREAD

**Issue by lahwaacz** [mpi4py/mpi4py#457]
I've built mpi4py 3.1.5 with OpenMPI 5.0.1 for Arch Linux and received the following error when running the test suite:
```
======================================================================
ERROR: testNamePublishing (test_dynproc.TestDPM.testNamePublishing)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/python-mpi4py/src/mpi4py-3.1.5/test/test_dynproc.py", line 57, in testNamePublishing
    MPI.Publish_name(service, port)
  File "mpi4py/MPI/Comm.pyx", line 2755, in mpi4py.MPI.Publish_name
mpi4py.MPI.Exception: MPI_ERR_INTERN: internal error

----------------------------------------------------------------------
Ran 1402 tests in 4.657s

FAILED (errors=1, skipped=117)
```

Is this a problem of mpi4py or openmpi? Any pointers how to investigate this closer?

**Comment by dalcinl** [comment]
It is most likely not an mpi4py problem. I may be Open MPI problem, although I cannot tell for sure, it could be a system configuration affecting Open MPI.  You should report the problem to the Arch maintainers for the Open MPI package, they may know better. Open MPI folks may be able to provide additional debugging tips (i.e. MCA parameters you can set to turn out verbose output).

**Comment by lahwaacz** [comment]
I've noticed the test succeeds when executed with `mpiexec` (even with `-n 1`), but fails when started without `mpiexec`.

**Comment by dalcinl** [comment]
mpi4py 3.1.5 was leased before Open MPI 5.0.0. Open MPI breaks mpi4py testsuite after every subminior/patch release. Right now, Open MPI folks are working on integrating mpi4py to their CI tests to improve their own testing.
Additionally, running without mpiexec used to break many MPI things with both MPICH and Open MPI. Therefore, running mpi4py tests without mpiexec was something I was not paying to much attention while testing on CI.  I've asked both MPICH and OMPI dev teams to improve on this situation, then they did the work the point that most things now work. Therefore, I started testing singleton init mode (that is,  running w/o mpiexec) in CI, but that only happened in mpi4py@master. However, if you look right now at the failing test, you will notice that these test are skip if run in less than 2 MPI processes (see commit 0b56f2909234aeacd86e69f94dcc6a9aabab0f54) https://github.com/mpi4py/mpi4py/blob/master/test/test_dynproc.py#L37. On a second thought, technically, I should not have skip the name publishing test. As the test is being skip, the non-mpiexec case is not currently covered. Anyway, doing name publishing without a process manager (i.e. mpiexec) is hard to support out of the box, so no surprises Open MPI is failing.

Long story short 1: If you run mpi4py testsuite and you get an MPI error, it is a 99% chance the issue comes from the MPI implementation, either from a bug or something the implementation doesn't support/have not implemented.

Long story short 2: Do not run mpi4py testsuite without mpiexec, as some tests that are know failures will run anyway.
